lib/puppet/indirector/facts/facter.rb

```diff
@@ -32,8 +32,13 @@ def find(request)
     # Initialize core Puppet facts, such as puppetversion
     Puppet.initialize_facts
 
-    result = Puppet::Node::Facts.new(request.key, Facter.to_hash)
-    result.add_local_facts
+    result = if request.options[:show]
+               find_with_options(request)
+             else
+               Puppet::Node::Facts.new(request.key, Facter.to_hash)
+             end
+
+    result.add_local_facts unless request.options[:show]
```

Comment:
Why exclude local facts?

Reply:
We wanted `puppet facts show` output to be identical to `facter -p` and the later command doesn't have the local facts. Since it's only an addition you are right, there won't be a problem if the facts will be added.